arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/CircuitBreaker.kt

```diff
@@ -40,10 +41,26 @@ class CircuitBreaker constructor(
       is State.HalfOpen -> curr.awaitClose.get()
     }
 
-  /** Returns a new task that upon execution will execute the given
+  /**
+   * Returns a new task that upon execution will execute the given
    * task, but with the protection of this circuit breaker.
    */
-  tailrec suspend fun <A> protect(fa: suspend () -> A): A =
+  @Deprecated("#protect is deprecated in favor of protectOrThrow", ReplaceWith("protectOrThrow"))
+  suspend fun <A> protect(fa: suspend () -> A): A =
+    protectOrThrow(fa)
+
+  /**
+   * Returns a new task that upon execution will execute the given
+   * task, but with the protection of this circuit breaker.
+   * If an exception in [fa] occurs it will be rethrown
+   */
+  suspend fun <A> protectOrThrow(fa: suspend () -> A): A =
+    protectEither(fa).fold({ throw it }, ::identity)
+
+  suspend fun <A> protectOrNull(fa: suspend () -> A): A? =
+    protectEither(fa).fold({ null }, ::identity)
+
+  tailrec suspend fun <A> protectEither(fa: suspend () -> A): Either<Throwable, A> =
```

Comment:
This should be `protectOrThrow`, and `protectEither` should wrap in `Either.catch`. That way we can avoid the allocations in the base implementation. `protectOrNull` can just `try/catch` and return `null`.